R/secret.R

```diff
@@ -113,6 +117,32 @@ secret_read_rds <- function(path, key) {
   secret_unserialize(x, key)
 }
 
+#' @export
+#' @param envir The unencrypted file will be automatically deleted when
+#'   this environment exits. You should only need set this argument if you
+#'   want to pass the unencrypted file to another function.
+#' @rdname secrets
+secret_decrypt_file <- function(path, key, envir = parent.frame()) {
+  val <- readChar(path, file.info(path)$size)
```

Comment:
Is there a particular reason for focusing on text files? I think httr2's focus on encrypting/decrypting strings and text is part of how I ended up writing `gargle::secret_(write|read)_rds()` and `gmailr::gm_token_(write|read)`.

Reply:
This is just because `secret_encrypt()` returns a base64 encoded string. This isn't really necessary for files, but it seems simple enough to keep it. It shouldn't affect your ability to encrypt and decrypt binary files. (And there's already `secret_write_rds()` and `secret_read_rds()` if you want rds files).

Reply:
Oh but I still need to read a binary file when encrypting. So probably better to be consistent.

Reply:
And that made me realise that there's a bunch of duplicate code between string and rds encryption, so overall a nice refactoring. Thanks for spotting the problem!